File: bpa-security/src/main/java/id/co/oriza/bpa/security/domain/model/Group.java
package id.co.oriza.bpa.security.domain.model;

import id.co.oriza.bpa.base.domain.model.ConcurrencySafeEntity;

public class Group extends ConcurrencySafeEntity {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	private String code;
	private String name;
	private String description;

	protected Group() {
		super();
	}

	public Group(String code, String name, String description) {
		this();
		this.setCode(code);
		this.setName(name);
		this.setDescription(description);
	}

	protected void setName(String name) {
		this.assertArgumentNotEmpty(name, "Name is required");
		this.name = name;
	}

	public String code() {
		return code;
	}

	public String name() {
		return name;
	}

	public String description() {
		return description;
	}

	protected void setCode(String code) {
		this.code = code;
	}

	protected void setDescription(String description) {
		this.description = description;
	}
	
	public void changeName(String aName){
		this.setName(aName);
	}

}

File: bpa-security-web/src/main/webapp/scripts/app/bpa/security/model/Group.ts
/**
 * Created by Yusuf on 5/25/2015.
 */

import Model = require("bpa/base/model/Model");

class Group extends Model{

    code: string;
    name: string;
    description: string;

    constructor(theCode: string, theName: string, theDescription: string){
        super();

        this.code = theCode;
        this.name = theName;
        this.description = theDescription;
    }

    static newInstance(): Group{
        return new Group(null,  null, null);
    }
}

export = Group;
